packages/minimongo/minimongo.js

```diff
@@ -221,14 +221,26 @@ LocalCollection.Cursor.prototype.fetch = function () {
  * @locus Anywhere
  * @returns {Number}
  */
-LocalCollection.Cursor.prototype.count = function () {
+LocalCollection.Cursor.prototype.count = function (applySkipLimit) {
   var self = this;
+  // Have to respect `applySkipLimit`, so we remove limit & skip if `applySkipLimit`
```

Comment:
I want to clear this up just in case. I know you know infinitely more about this, so treading lightly here. I believe Minimongo by design is supposed to return immediately and therefore there shouldn't be any Fiber/Future/Promise logic and modifying `self.limit` and `self.skip` on the go should be fine there. I've read the Minimongo source way too much by now.

Over at the server side, we just pass this problem over to the [MongoDB node module](https://www.npmjs.com/package/mongodb). It supports `applySkipLimit` and we don't have to deal with the problem.

Is this not an acceptable solution?


What would a smooth upgrade path look like? On the server, nothing changes, `count()` works exactly the same as before. On client side, `count()` would now have the opposite result (before, it worked as if `applySkipLimit` defaulted to `true`, now it would default to `false`).